src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -515,6 +516,14 @@ export const createXMLHttpRequestOverride = (
       }
     }
 
+    getResponseXML() {
+      const contentType = this.getResponseHeader('Content-Type')
+      if (contentType === 'application/xml' || contentType === 'text/xml') {
+        return new DOMParser().parseFromString(this.responseText, contentType)
```

Comment:
FYI, in the event of an invalid XML string, `parseFromString` will return a `Document` representing an error state. See https://developer.mozilla.org/en-US/docs/Web/API/DOMParser/parseFromString#error_handling.